src/main/kotlin/org/virtuslab/pulumikotlin/codegen/step3codegen/resources/ResourceGenerator.kt

```diff
@@ -137,6 +139,9 @@ object ResourceGenerator {
             .addDocs("@param block The arguments to use to populate this resource's properties.")
             .build()
 
+        requireNotNull(::opts)
```

Comment:
I wanted to make this compile-time safe and I couldn't find a way to interpolate a method reference in KotlinPoet

Reply:
Ok, if it reduces the risk of generating non-compiling code then should be fine. Maybe you could drop some code comment here?

---

(kinda-related)

I remember thinking about writing some custom method to convert Kotlin method reference into `MemberName`. 

Possible signatures `(::opts).toMemberName()`, `memberNameFrom(::opts)`. 

It's tricky with generics and overloads, though. I'm also not sure if extracting package/class is possible from `KFunction`.

Reply:
I tried to figure out a way to do that with `KFunction`, but I couldn't get it to work :/